### ISSUE TITLE
Bump versions to eliminate CVEs: golang.org/x/net golang.org/x/net

### ISSUE THREAD

**Issue by seanorama** [enix/x509-certificate-exporter#210]
Version bump needed to eliminate CVEs:

```
❯ trivy image enix/x509-certificate-exporter:3.10.1
2023-11-15T20:05:15.824-0600	INFO	Vulnerability scanning is enabled
2023-11-15T20:05:15.824-0600	INFO	Secret scanning is enabled
2023-11-15T20:05:15.824-0600	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-11-15T20:05:15.824-0600	INFO	Please see also https://aquasecurity.github.io/trivy/v0.45/docs/scanner/secret/#recommendation for faster secret detection
2023-11-15T20:05:19.567-0600	INFO	Number of language-specific files: 1
2023-11-15T20:05:19.567-0600	INFO	Detecting gobinary vulnerabilities...

x509-certificate-exporter (gobinary)

Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 1, CRITICAL: 0)

┌──────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│     Library      │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├──────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/net │ CVE-2023-39325 │ HIGH     │ fixed  │ v0.15.0           │ 0.17.0        │ golang: net/http, x/net/http2: rapid stream resets can cause │
│                  │                │          │        │                   │               │ excessive work (CVE-2023-44487)                              │
│                  │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2023-39325                   │
│                  ├────────────────┼──────────┤        │                   │               ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2023-44487 │ MEDIUM   │        │                   │               │ HTTP/2: Multiple HTTP/2 enabled web servers are vulnerable   │
│                  │                │          │        │                   │               │ to a DDoS attack...                                          │
│                  │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2023-44487                   │
└──────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘
```

Also, it appears `latest` on docker hub is pointing to an old version as it returns this:

```
❯ trivy image enix/x509-certificate-exporter
2023-11-15T19:44:33.051-0600	INFO	Vulnerability scanning is enabled
2023-11-15T19:44:33.052-0600	INFO	Secret scanning is enabled
2023-11-15T19:44:33.052-0600	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-11-15T19:44:33.052-0600	INFO	Please see also https://aquasecurity.github.io/trivy/v0.45/docs/scanner/secret/#recommendation for faster secret detection
2023-11-15T19:44:36.120-0600	INFO	Number of language-specific files: 1
2023-11-15T19:44:36.120-0600	INFO	Detecting gobinary vulnerabilities...

x509-certificate-exporter (gobinary)

Total: 6 (UNKNOWN: 0, LOW: 0, MEDIUM: 2, HIGH: 4, CRITICAL: 0)

┌───────────────────┬────────────────┬──────────┬────────┬────────────────────────────────────┬─────────────────────────────────────┬──────────────────────────────────────────────────────────────┐
│      Library      │ Vulnerability  │ Severity │ Status │         Installed Version          │            Fixed Version            │                            Title                             │
├───────────────────┼────────────────┼──────────┼────────┼────────────────────────────────────┼─────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/net  │ CVE-2022-41721 │ HIGH     │ fixed  │ v0.0.0-20220909164309-bea034e7d591 │ 0.1.1-0.20221104162952-702349b0e862 │ request smuggling                                            │
│                   │                │          │        │                                    │                                     │ https://avd.aquasec.com/nvd/cve-2022-41721                   │
│                   ├────────────────┤          │        │                                    ├─────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                   │ CVE-2022-41723 │          │        │                                    │ 0.7.0                               │ net/http, golang.org/x/net/http2: avoid quadratic complexity │
│                   │                │          │        │                                    │                                     │ in HPACK decoding                                            │
│                   │                │          │        │                                    │                                     │ https://avd.aquasec.com/nvd/cve-2022-41723                   │
│                   ├────────────────┤          │        │                                    ├─────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                   │ CVE-2023-39325 │          │        │                                    │ 0.17.0                              │ golang: net/http, x/net/http2: rapid stream resets can cause │
│                   │                │          │        │                                    │                                     │ excessive work (CVE-2023-44487)                              │
│                   │                │          │        │                                    │                                     │ https://avd.aquasec.com/nvd/cve-2023-39325                   │
│                   ├────────────────┼──────────┤        │                                    ├─────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                   │ CVE-2023-3978  │ MEDIUM   │        │                                    │ 0.13.0                              │ golang.org/x/net/html: Cross site scripting                  │
│                   │                │          │        │                                    │                                     │ https://avd.aquasec.com/nvd/cve-2023-3978                    │
│                   ├────────────────┤          │        │                                    ├─────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                   │ CVE-2023-44487 │          │        │                                    │ 0.17.0                              │ HTTP/2: Multiple HTTP/2 enabled web servers are vulnerable   │
│                   │                │          │        │                                    │                                     │ to a DDoS attack...                                          │
│                   │                │          │        │                                    │                                     │ https://avd.aquasec.com/nvd/cve-2023-44487                   │
├───────────────────┼────────────────┼──────────┤        ├────────────────────────────────────┼─────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/text │ CVE-2022-32149 │ HIGH     │        │ v0.3.7                             │ 0.3.8                               │ golang: golang.org/x/text/language: ParseAcceptLanguage      │
│                   │                │          │        │                                    │                                     │ takes a long time to parse complex tags                      │
│                   │                │          │        │                                    │                                     │ https://avd.aquasec.com/nvd/cve-2022-32149                   │
└───────────────────┴────────────────┴──────────┴────────┴────────────────────────────────────┴─────────────────────────────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by seanorama** [comment]
Discovered that dependabot has suggested this change in #198

**Comment by npdgm** [comment]
Hi @seanorama
Thanks a lot for raising those issues, and sorry for the time it took to fix CVEs. I was traveling but the team could have picked it up.
So version 3.11.0 was released with an upgrade to latest dependencies and Go version. Trivy scans are clean now.
As for the issue with `latest`, it looks like we have a CI problem. The tag was fixed manually for 3.11.0 and I'll be investigating the CI so it gets updated later.
